solr/core/src/java/org/apache/solr/handler/RequestHandlerBase.java

```diff
@@ -250,17 +253,17 @@ public void handleRequest(SolrQueryRequest req, SolrQueryResponse rsp) {
       long elapsed = timer.stop();
       metrics.totalTime.inc(elapsed);
 
-      if (cpuStats != null) {
-        Optional<Long> cpuTime = cpuStats.getCpuTimeMs();
+      if (publishCpuTime) {
+        Optional<Long> cpuTime = threadCpuTime.getCpuTimeMs();
         if (cpuTime.isPresent()) {
           // add CPU_TIME if not already added by SearchHandler
           NamedList<Object> header = rsp.getResponseHeader();
```

Comment:
I see lots of places where this is checked for null, but I'm wondering why it's allowed to return null (not a topic  for this ticket however)

Reply:
Yeah, I don't like this either. `ThreadCpuTimer.isSupported()` is the only condition for this to return false and it should be enough to check it.